pkg/trace/traceutil/otel_util.go

```diff
@@ -205,8 +205,140 @@ func GetOTelResource(span ptrace.Span, res pcommon.Resource) (resName string) {
 	return
 }
 
-// GetOTelOperationName returns the DD operation name based on OTel span and resource attributes and given configs.
-func GetOTelOperationName(
+// GetOTelResourceV2 returns the DD resource name based on OTel span and resource attributes.
+func GetOTelResourceV2(span ptrace.Span, res pcommon.Resource) (resName string) {
+	resName = GetOTelAttrValInResAndSpanAttrs(span, res, false, "resource.name")
+	if resName == "" {
+		if m := GetOTelAttrValInResAndSpanAttrs(span, res, false, "http.request.method", semconv.AttributeHTTPMethod); m != "" {
+			if m == "_OTHER" {
+				m = "HTTP"
+			}
+			// use the HTTP method + route (if available)
+			resName = m
+			if span.Kind() == ptrace.SpanKindServer {
+				if route := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeHTTPRoute); route != "" {
+					resName = resName + " " + route
+				}
+			}
+		} else if m := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeMessagingOperation); m != "" {
+			resName = m
+			// use the messaging operation
+			if dest := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeMessagingDestination, semconv117.AttributeMessagingDestinationName); dest != "" {
+				resName = resName + " " + dest
+			}
+		} else if m := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeRPCMethod); m != "" {
+			resName = m
+			// use the RPC method
+			if svc := GetOTelAttrValInResAndSpanAttrs(span, res, false, semconv.AttributeRPCService); m != "" {
+				// ...and service if available
+				resName = resName + " " + svc
+			}
+		} else {
+			resName = span.Name()
+		}
+	}
+	if len(resName) > MaxResourceLen {
+		resName = resName[:MaxResourceLen]
```

Comment:
Rewrite this to avoid if/else blocks. For ex 
```
if resName != "" { 
return resName[:MaxResourceLen]
}
```

Reply:
Wouldn't this give an out of bounds error if `len(resName) < MaxResourceLen`?